src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java

```diff
@@ -75,13 +77,31 @@ public AcSensitivityAnalysis(MatrixFactory matrixFactory) {
         return sensitivityValues;
     }
 
-    protected void setReferenceActivePowerFlows(List<LfSensitivityFactor<ClosedBranchSide1ActiveFlowEquationTerm>> factors) {
-        for (LfSensitivityFactor factor : factors) {
-            factor.setFunctionReference(factor.getFunctionLfBranch().getP1());
+    private Class<? extends AbstractClosedBranchAcFlowEquationTerm> getEquationTermClass(SensitivityFactor factor) {
+        if (factor.getFunction() instanceof BranchFlow) {
+            return ClosedBranchSide1ActiveFlowEquationTerm.class;
+        } else if (factor.getFunction() instanceof BranchIntensity) {
+            return ClosedBranchSide1IntensityMagnitudeEquationTerm.class;
+        } else {
+            throw new NotImplementedException("Unknown function type: " + factor.getFunction().getClass().getSimpleName());
         }
     }
 
-    private List<SensitivityValue> getPostContingencySensitivityValues(List<LfSensitivityFactor<ClosedBranchSide1ActiveFlowEquationTerm>> lfFactors,
+    protected void setReferenceActivePowerFlows(List<LfSensitivityFactor<? extends AbstractClosedBranchAcFlowEquationTerm>> factors) {
```

Comment:
You have to change the method name, and go back to one of your first choices `setReferenceFunctions` maybe?